types/time.go

```diff
@@ -21,8 +21,7 @@ func (t *Time) UnmarshalJSON(data []byte) error {
 		s = s[1 : len(s)-1]
 	}
 
-	switch s {
-	case "null", "0", "":
+	if s == "" || s[0] == 'n' || (s[0] == '0' && strings.Trim(s, "0.") == "") {
```

Comment:
🚧  So we're saying that if it's `0.0000` then it's nil. That's okay.
But we'd also be parsing `0.0.0.0` as nil, and I think that should be an error condition.

So I'd be tempted to move this after L28, remove a single `.` from the input, and then here we can check if it's just 0s with a `TrimPrefix(s, "0") == ""` ( or any other method).
That way we'd still error on L42 if we get whacky input that's probably an IP address or something, by mistake.